### PR TITLE
v1.43.0 changelog updates

### PR DESCRIPTION
**Pull request by displague**
* Adds Changelog.md items with all of the changes since v1.42.0 was released
* Adds RELEASE.md (borrowing from setup.py) to describe the release process

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,16 +4,28 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
-## [1.43.0] - 2020-02-24
+## [1.44.0] - Unreleased
+### Added
+### Changed
+### Fixed
+
+## [1.43.0] - 2020-07-14
```
Comment (mmlb) [inline CHANGELOG.md:12]: we need to update the version number in [setup.py](https://github.com/packethost/packet-python/blob/f32e55174b43130e6782828d0fbc712ba644ac92/setup.py#L51) and [packet/_init_.py](https://github.com/packethost/packet-python/blob/f32e55174b43130e6782828d0fbc712ba644ac92/packet/__init__.py#L6).

We should add this to the RELEASE.md too

Comment (displague) [reply CHANGELOG.md:12]: I've updated the RELEASE.md process and changed `setup.py` to pull the version from `packet/__init__.py`.

Comment (displague) [reply CHANGELOG.md:12]:
```
$ python2 ./setup.py --version
1.43.0
$ python3 ./setup.py --version
1.43.0
```

Comment (displague) [reply CHANGELOG.md:12]: I used the first recommendation from https://packaging.python.org/guides/single-sourcing-package-version/

**Pull request merged**